[master<20230420] Improve backup and restore docs (#828)
memgraph/docs: docs/how-to-guides/create-backup.md

data in a recoverable format. On startup, it searches for previously saved
durability files and uses them to recreate the most recent DB state.

When talking about the data directory, we are actually talking about two
directories, `snapshots` and `wal` which are usually located in the
`/var/lib/memgraph` folder.

> 

Reviewer: I think in general the data directory is the directory that is specified as the `--data-directory`, which contains other directories, such as `snapshots`, `wal`, but other directories also, `streams`, `triggers`, etc.

Author: i can add
```suggestion
When talking about the data directory in the context of backup and restore, we are actually talking about two
directories, `snapshots` and `wal` which are usually located in the
`/var/lib/memgraph` folder.

```

Author: @antaljanosbenjamin what do you think about this change?

Reviewer: Perfect!

### 1. Lock the data directory

> To disable file deletions in the data directory, run the following query in

Reviewer: ```suggestion
To disable deletions of snapshot or WAL files in the data directory, run the following query in
```

`mgconsole` or Memgraph Lab:

```cypher
LOCK DATA DIRECTORY;
```

### 2. Copy the snapshot into the data directory

Copy the snapshot or WAL directory or, empty it and then copy a single WAL or
snapshot file into it.

If you need help copying the files from the Docker container, check out the
[Working with docker
guide](/how-to-guides/work-with-docker.md#how-to-copy-files-from-and-to-a-docker-container).

### 3. Restart the instance

By restarting the instance, Memgraph should restore the data from the files in
the data directory. 

> Be sure to restart the instance before Memgraph automatically creates a new

Author: ```suggestion
If you unlock the data directory, be sure to restart the instance before Memgraph automatically creates a new
```

> periodic snapshot because upon restart it will use that newer snapshot which is
> not the data you want to load. 

Reviewer: ```suggestion
periodic snapshot because upon restart it might use that newer snapshot which is
not the data you want to load. 
```

It depends on the lexicographically order of them. By default the files are named by the datetime they are created (something like `20230420132211347350_timestamp_1`), so if the users use the default names (really just copying the files without renaming), then what you wrote is totally true.
